front/app/modules/commercial/insights/admin/containers/Insights/Details/Network/index.tsx

```diff
@@ -144,35 +165,144 @@ const Network = ({
       ctx.textAlign = 'center';
       ctx.textBaseline = 'middle';
       ctx.fillStyle = nodeColors[node.color_index % nodeColors.length];
+      const textWidth = ctx.measureText(label).width;
+      node.nodeVerticalOffset = nodeVerticalOffset;
+      node.textWidth = textWidth;
+      node.globalScale = globalScale;
+      node.nodeFontSize = nodeFontSize;
 
       if (globalScale >= visibleKeywordLabelScale) {
         ctx.fillText(label, node.x, nodeVerticalOffset);
+
+        if (node == hoverNode) {
+          // Draw hide icon
+          const hideIcon = new Path2D(
```

Comment:
I wonder if we could find a way to take some of these out of the component and make it a bit more readable.